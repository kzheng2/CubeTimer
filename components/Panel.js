import React from 'react'
import { inject, observer } from 'mobx-react'

@inject('recordsStore')
@observer
class Panel extends React.Component {
  render () {
    const { recordsStore: { clear, records } } = this.props
    if (records.length < 1) {
      return null
    }
    return (
      <div className='timebar' style={{
        display: 'flex',
        justifyContent: 'center'
      }}>
        <span onClick={clear}>
          clear
        </span>
      </div>
    )
  }
}

export default Panel
