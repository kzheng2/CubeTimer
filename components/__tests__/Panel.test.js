/* global it, expect, describe */
import React from 'react'
import Panel from '../Panel'
import { spy } from 'sinon'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Panel', () => {
  it('render null if records is empty', () => {
    const recordsStore = { records: [] }
    const el = shallow(<Panel recordsStore={recordsStore} />)
    expect(el).toBeTruthy()
    expect(el.text()).toBe('')

    const tree = renderer.create(
      <Panel recordsStore={recordsStore} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render clear and call clear onClick', () => {
    const recordsStore = { records: [ 42 ], clear: spy() }
    const expectedText = 'clear'
    const el = shallow(<Panel recordsStore={recordsStore} />)
    expect(el.text()).toBe(expectedText)
    expect(el.find({ children: expectedText }).exists()).toBe(true)
    expect(recordsStore.clear.callCount).toBe(0)
    el.find({ children: expectedText }).props().onClick()
    expect(recordsStore.clear.callCount).toBe(1)
    el.find({ children: expectedText }).props().onClick()
    expect(recordsStore.clear.callCount).toBe(2)

    const tree = renderer.create(
      <Panel recordsStore={recordsStore} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
