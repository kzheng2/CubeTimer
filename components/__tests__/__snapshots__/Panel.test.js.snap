// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel render clear and call clear onClick 1`] = `
<div
  className="timebar"
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <button
    className="MuiButtonBase-root-15 MuiButton-root-1"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    role="button"
    style={
      Object {
        "width": "100%",
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-3"
    >
      Clear
    </span>
    <span
      className="MuiTouchRipple-root-17"
    />
  </button>
</div>
`;

exports[`Panel render null if records is empty and stopwatch is running or lapse is 0 1`] = `null`;
